crates/api/src/v1/monitor.rs

```diff
@@ -0,0 +1,124 @@
+//! Types relating to the monitor API.
+
+use serde::{Deserialize, Serialize, Serializer};
+use std::borrow::Cow;
+use thiserror::Error;
+
+/// Represents checkpoint verification response.
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CheckpointVerificationResponse {
+    /// The checkpoint verification state.
+    pub checkpoint: CheckpointVerificationState,
+    /// The checkpoint signature verification state.
+    pub signature: CheckpointSignatureVerificationState,
+}
+
+/// Represents checkpoint verification state.
+#[derive(Eq, PartialEq, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum CheckpointVerificationState {
+    /// The checkpoint signature is unverified and could be valid or invalid.
+    #[serde(rename_all = "camelCase")]
+    Unverified,
+    /// The checkpoint is verified.
+    #[serde(rename_all = "camelCase")]
+    Verified,
+    /// The checkpoint log length does not exist.
+    #[serde(rename_all = "camelCase")]
+    NotFound,
+    /// The checkpoint is invalid.
+    #[serde(rename_all = "camelCase")]
+    Invalid,
+}
+
+/// Represents checkpoint signature verification state.
+#[derive(Eq, PartialEq, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum CheckpointSignatureVerificationState {
+    /// The checkpoint signature is unverified and could be valid or invalid.
+    #[serde(rename_all = "camelCase")]
+    Unverified,
+    /// The checkpoint signature is verified.
+    #[serde(rename_all = "camelCase")]
+    Verified,
+    /// The checkpoint signature key ID is known but not authorized to sign checkpoints.
+    #[serde(rename_all = "camelCase")]
+    Unauthorized,
+    /// The checkpoint signature is not valid. The key ID could be not known or the signature is incorrect.
+    #[serde(rename_all = "camelCase")]
+    Invalid,
+}
+
+/// Represents a monitor API error.
+#[non_exhaustive]
+#[derive(Debug, Error)]
+pub enum MonitorError {
+    /// Instruct to retry after specified number of seconds.
+    #[error("retry after {0} seconds")]
+    RetryAfter(u16),
```

Comment:
This still seems to be arbitrarily splitting states between OK and error responses. "Retry after" seems like it should be an optional bit of data associated with the `unverified` (or `notFound`?) state.

Reply:
Updated, from our discussion